core/src/main/java/org/elasticsearch/cluster/routing/IndexRoutingTable.java

```diff
@@ -94,7 +94,7 @@ public Index getIndex() {
         return index;
     }
 
-    public void validate(RoutingTableValidation validation, MetaData metaData) {
+    void validate(RoutingTableValidation validation, MetaData metaData) {
```

Comment:
instead of accumulating validations under RoutingTableValidation, we can throw an exception immediately upon encountering an error. This will remove one more class and will simplify things further.